File: MyApplication2/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;




import android.os.Bundle;



import android.view.View;

import android.widget.Button;

import android.widget.DatePicker;

import android.widget.TextView;


import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.DialogFragment;

import java.text.DateFormat;

import java.util.Calendar;



public class MainActivity extends AppCompatActivity implements DatePickerFragment.DatePickerListener {



    private TextView tvDisplayDate;





    @Override

    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);



        tvDisplayDate = findViewById(R.id.tvDisplaydate);



        Button btnShowDatePicker = findViewById(R.id.btnShowDatePicker);

        btnShowDatePicker.setOnClickListener(new View.OnClickListener() {

            @Override

            public void onClick(View view) {

                DialogFragment datePickerFragment=new DatePickerFragment();

                datePickerFragment.setCancelable(false);

                datePickerFragment.show(getSupportFragmentManager(),"date Picker");

            }

        });



    }



    @Override

    public void onDateSet(DatePicker datePicker, int year, int month, int day) {

        Calendar cal=Calendar.getInstance();

        cal.set(Calendar.YEAR,year);

        cal.set(Calendar.MONTH,month);

        cal.set(Calendar.DAY_OF_MONTH,day);

        String date= DateFormat.getDateInstance().format(cal.getTime());

        tvDisplayDate.setText(date);



    }

}

File: Datekt/app/src/main/java/com/example/datekt/MainActivity.kt
package com.example.datekt

import android.os.Bundle
import android.widget.Button
import android.widget.DatePicker
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.DialogFragment
import com.example.datekt.DatePickerFragment.DatePickerListener
import java.text.DateFormat
import java.util.*

class MainActivity : AppCompatActivity(), DatePickerListener {
    private var tvDisplayDate: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tvDisplayDate = findViewById(R.id.tvDisplaydate)
        val btnShowDatePicker =
            findViewById<Button>(R.id.btnShowDatePicker)
        btnShowDatePicker.setOnClickListener {
            val datePickerFragment: DialogFragment = DatePickerFragment()
            datePickerFragment.isCancelable = false
            datePickerFragment.show(supportFragmentManager, "date Picker")
        }
    }

    override fun onDateSet(
        datePicker: DatePicker?,
        year: Int,
        month: Int,
        day: Int
    ) {
        val cal = Calendar.getInstance()
        cal[Calendar.YEAR] = year
        cal[Calendar.MONTH] = month
        cal[Calendar.DAY_OF_MONTH] = day
        val date = DateFormat.getDateInstance().format(cal.time)
        tvDisplayDate!!.text = date
    }
}
